File: advent2a.py
ans = 0
with open("input.txt", "r") as fp:
    for line in fp.readlines():
        levels = [int(level) for level in line.split(" ")]
        if len(levels) < 2:
            ans += 1
            continue

        increasing = levels[0] < levels[1]
        good = True
        for i in range(1,len(levels)):
            if increasing and (levels[i] - levels[i-1] <= 0 or levels[i] - levels[i-1] > 3):
                good = False
                break
            if not increasing and (levels[i] - levels[i-1] >= 0 or levels[i] - levels[i-1] < -3):
                good = False
                break
        if good:
            ans += 1

print(ans)

File: aoc2025/2a/advent2a.java
import java.io.File;
import java.util.ArrayList;
import java.util.Scanner;

public class advent2a {
    record Range(long left, long right) {
    }

    static private ArrayList<Range> ranges = new ArrayList<>();

    public static long solution() {
        long ans = 0L;
        for (Range range : ranges) {
            System.out.println(range);

            String leftString = String.valueOf(range.left);
            String rightString = String.valueOf(range.right);

            for (int size = leftString.length(); size <= rightString.length(); size++) {
                if (size % 2 != 0) continue;

                long repeatable = (long)Math.pow(10, (size/2)-1);
                while (repeatable < Math.pow(10, size/2)) {
                    long id = repeatable + repeatable * (long)Math.pow(10, size/2);

                    repeatable++;
                    if (id < range.left) continue;
                    if (id > range.right) break;

                    System.out.println(id);
                    ans += id;
                }
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        try {
            File fp = new File("input.txt");
            Scanner fpr = new Scanner(fp);

            for (String rangeString : fpr.nextLine().split(",")) {
                long left = Long.parseLong(rangeString.split("-")[0]);
                long right = Long.parseLong(rangeString.split("-")[1]);
                ranges.add(new Range(left, right));
            }

            fpr.close();

            System.out.println(solution());

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
